Allow keystore add to handle multiple settings

jasontedor opened the pull request:
Today the keystore add command can only handle adding a single setting/value pair in a single invocation. This incurs the startup costs of the JVM many times, which in some environments can be expensive. This commit teaches the add keystore command to accept adding multiple settings in a single invocation.

Closes #54191

jasontedor committed:
Allow keystore add to handle multiple settings

Today the keystore add command can only handle adding a single
setting/value pair in a single invocation. This incurs the startup costs
of the JVM many times, which in some environments can be expensive. This
commit teaches the add keystore command to accept adding multiple
settings in a single invocation.

elasticmachine commented:
Pinging @elastic/es-core-infra (:Core/Infra/Core)

jasontedor committed:
Fix checkstyle

jasontedor committed:
Merge remote-tracking branch 'elastic/master' into keystore-add-multiple-settings

pugnascotia commented on distribution/tools/keystore-cli/src/main/java/org/elasticsearch/common/settings/AddStringKeyStoreCommand.java:
```diff
@@ -42,13 +46,13 @@
     AddStringKeyStoreCommand() {
-        super("Add a string setting to the keystore", false);
-        this.stdinOption = parser.acceptsAll(Arrays.asList("x", "stdin"), "Read setting value from stdin");
+        super("Add string settings to the keystore", false);
```
Would it be worth updating the equivalent text in `RemoveSettingKeyStoreCommand`? It still talks about removing "a setting" from the keystore, but it already supports multiple settings.

jasontedor replied:
I will handle this in a follow up.

jasontedor replied:
I opened #54244.

pugnascotia commented on docs/reference/commands/keystore.asciidoc:
```diff
@@ -70,12 +71,14 @@ protected, you are prompted to enter the current password and the new one. You
 
-`--stdin`:: When used with the `add` parameter, you can pass the setting value
-through standard input (stdin). See <<add-string-to-keystore>>.
+`--stdin`:: When used with the `add` parameter, you can pass the settings values
```
Should we document `-x` too?

pugnascotia replied:
I notice we only have `-f` without `--force` as well.

jasontedor replied:
Thanks for noticing these. Since these are orthogonal to this change, I will handle them in a follow up.

jasontedor replied:
I opened #54242.

bpintea commented:
7.7 being feature frozen, I was wondering if the `v7.7.0` label should be removed,  or if this should be treated as a blocker or a fix -- considering the tracking for the release notes.

rjernst commented on docs/reference/commands/keystore.asciidoc:
```diff
@@ -40,12 +40,13 @@ keystore, see the setting reference.
+`add <settings>`:: Adds settings to the keystore. Multiple setting names can be
+specified as arguments to the `add` command. By default, you are prompted for
+the values of the settings. If the keystore is password protected, you are also
+prompted to enter the password. If the setting already exists in the keystore,
```
the setting -> a setting

rjernst reviewed (approved):
LGTM

jasontedor committed:
Fix docs

jasontedor commented:
@bpintea This is a late addition to the feature list for 7.7.0 because it's aimed at addressing some issues on Cloud, and we don't want to wait for a fix for those until 7.8.0, which would delay it being in their hands for awhile.
